graphqlmetrics/data_migrations/add_span_id_to_traces.md

```diff
@@ -0,0 +1,237 @@
+# New SpanID column to traces table to accurately retrieve trace information
+
+## Purpose:
+This migration will help customers to retrieve accurate trace information. This can happen when there are multiple root spans for a given Trace ID.
+
+### Migration Overview:
+We need to drop both the `traces_mv` materialized view and `traces` table, recreate and repopulate them. We do the repopulation in hourly batches.
+
+---
+
+## Prerequisites:
+Before proceeding, ensure the following:
+
+1. **If you need to do this**: If you already have SpanID in `traces` and `traces_mv` then you can skip this entire migration.
+2. **Back up existing data**: Always back up your existing tables and data to avoid any accidental data loss.
+3. **Database maintenance window**: Schedule downtime if necessary to avoid service interruptions during the migration.
+
+---
+
+## Migration Steps
+
+Please ensure the latest version of Cosmo has been deployed. This migration is only applicable to the following components:
+
+- **Clickhouse**: 24.4
+- **Controlplane**: vx.Y
+
+### Step 1: Drop view and table
+
+It is good practice to drop the materialized view first
+
+```sql
+DROP VIEW IF EXISTS cosmo.traces_mv;
+DROP TABLE IF EXISTS cosmo.traces;
+```
+
+### Step 2: Recreate the `traces` table only
+
+After running the above script, we need to recreate the table. We will create the materialized view again after repopulation.
+
+```sql
+CREATE TABLE IF NOT EXISTS traces (
```

Comment:
Let's not duplicate here if we need it anyway in the script. let's the scripts import the files once.